### PR TITLE
sync: Handle parse errors, make some preflight fields optional

### PR DESCRIPTION
**Pull request by russellhancox**

### CODE REVIEW

File: Source/santasyncservice/SNTSyncPreflight.mm [hunk 1]
```diff
@@ -135,7 +135,11 @@ - (BOOL)sync {
       resp.deprecated_transitive_whitelisting_enabled());
   self.syncState.enableAllEventUpload = @(resp.enable_all_event_upload());
   self.syncState.disableUnknownEventUpload = @(resp.disable_unknown_event_upload());
-  self.syncState.eventBatchSize = resp.batch_size();
+
+  self.syncState.eventBatchSize = 50;
```
Comment (mlw) [inline Source/santasyncservice/SNTSyncPreflight.mm:139]: Is this some known/previous default? Can we make it a constant and comment it?

**Review by mlw — CHANGES_REQUESTED** [review]

File: Source/santasyncservice/SNTSyncPreflight.mm (hunk 1 above)
Comment (russellhancox) [reply Source/santasyncservice/SNTSyncPreflight.mm]: Done

File: Source/santasyncservice/syncv1.proto [hunk 2]
```diff
@@ -54,7 +54,11 @@ message PreflightRequest {
 }
 
 message PreflightResponse {
```
Comment (pmarkowsky) [inline Source/santasyncservice/syncv1.proto:56]: Long as you're in here. We should add an optional machine_id field for GRPC.

Comment (russellhancox) [reply Source/santasyncservice/syncv1.proto:57]: As discussed, let's do this in a separate PR

**Pull request merged**